src/io/index.js

```diff
@@ -0,0 +1,308 @@
+                    const result = await this.createFileFromArrayBuffer(file);
+                    result.path = filePath;
+                    resolve(result);
+                } catch (e) {
+                    return reject(e.message);
+                }
+            };
+            handleUpload(blob);
+        });
+    }
+
+    // Generates a Data URI for the specified file
+    async getFileDataURL(path) {
+        return new Promise((resolve, reject) => {
+            const handle = async () => {
+                try {
+                    const fileInfo = await this.getFileInfo(path);
+                    var contents = fileInfo.body.contents;
+                    var name = fileInfo.body.name;
+                    var file = new File([contents], name, {
+                        type: fileInfo.type,
+                    });
+                    const fileDataURI = await this.readUploadedFileAsDataURL(
+                        file
+                    );
+                    resolve({ name: name, dataurl: fileDataURI });
+                } catch (e) {
+                    return reject(e.message);
+                }
+            };
+            handle();
+        });
+    }
+
+    // Uploads a file into filesystem
+    async createFileFromArrayBuffer(file) {
+        return new Promise((resolve, reject) => {
+            var buf = buffer(file.buffer);
+            fs.writeFile(file.path + file.name, buf, function(err) {
+                if (err) return reject({ success: false, err: err });
+                else resolve({ success: true });
+            });
+        });
+    }
+
+    // Uploads array of files into filesystem
+    async createFilesFromArrayBuffer(files) {
+        const promises = files.map(async file => {
+            file.buffer = Object.values(file.buffer);
+            const result = await this.createFileFromArrayBuffer(file);
+            return result;
+        });
+
+        return await Promise.all(promises);
+    }
+
+    // Deletes everything in Path
+    async deletePathRecursively(path) {
+        return new Promise((resolve, reject) => {
+            fs.stat(path, function(err, stats) {
+                if (err) return reject(format404(path));
+
+                // If this is a dir, begin recursive deletion
+                if (stats.isDirectory()) {
+                    sh.rm(path, { recursive: true }, function(err) {
+                        if (err) return reject(err);
+                        resolve(`Successfully removed ${path}.`);
+                    });
+                } else {
+                    fs.unlink(path, function(err) {
+                        if (err) return reject(err);
+                        resolve(`Successfully removed ${path}.`);
+                    });
+                }
+            });
+        });
+    }
+
+    // Deletes a given Directory /A
+    async deleteDirectory(path) {
+        return new Promise((resolve, reject) => {
+            fs.rmdir(path, function(err) {
```

Comment:
NOTE: this will require the dir to be empty.  Is that what you want?  If not, `sh.rm`